sds_data_manager/stacks/backup_bucket_stack.py

```diff
@@ -0,0 +1,96 @@
+from aws_cdk import (
+    Environment,
+    RemovalPolicy,
+    Stack,
+)
+from aws_cdk import (
+    aws_iam as iam,
+)
+from aws_cdk import (
+    aws_s3 as s3,
+)
+from constructs import Construct
+
+
+class BackupBucket(Stack):
+    """
+    This stack creates the destination bucket for data backups. It can be run in
+    the same account as SdsDataManager, or in a separate account.
+    The source_account is a required parameter. This source account should be the
+    AWS account for the source bucket.
+
+    For replication to work, you also need to deploy SdsDataManager and create
+    the source bucket and replication role. Then, you need to manually update
+    the role_arn variable with the replication role created.
+
+    Finally, you need to set up the replication rule in the **source** account.
+    To do this, go to the source bucket and click the "Management" tab.
+    Under the "Replication Rules" section, create a replication rule. Specify
+    your bucket, select the IAM Role "BackupRole" created in SdsDataManager, and
+    save the rule. This rule can be set up after both source and destination stacks
+    are deployed.
+    """
+
+    def __init__(
+        self,
+        scope: Construct,
+        construct_id: str,
+        sds_id: str,
+        env: Environment,
+        source_account: str,
+        **kwargs,
+    ) -> None:
+        """
+        BackupBucketStack
+
+        Parameters
+        ----------
+        scope : Construct
+        construct_id : str
+        sds_id: str
+        env : Environment
+            Account and region
+        source_account : str
+            Account number for the source S3 bucket
+        """
+        super().__init__(scope, construct_id, env=env, **kwargs)
+
+        # FOR NOW: Deploy other stack, update this name with the created role.
+        role_arn = (
+            "arn:aws:iam::449431850278:"
+            "role/SdsDataManager-mh-dev-BackupRoleF43CFD90-11WK8QFUD7HTG"
+        )
+
+        # This is the S3 bucket used by upload_api_lambda
+        backup_bucket = s3.Bucket(
```

Comment:
I see. We could add comment saying that this part is creating backup bucket in a different account to minimize confusion because they look exactly same when we look at it first. :)

or add suffix `backup`. You know how sds_id is set to `dev` or `prod`. Will `sds_id` be set to `backup` for backup account? If that's the case, then we are good.

A question. If we keep same exact bucket name convention for both source account and backup account, will it not give "bucket name is not unique" error?

Reply:
ignore my comments. You already do this!

Reply:
Yes, the sds_id is set to `backup` for the backup account, so all the deployed resources have "backup" in the name! Does that all make sense?